### ISSUE TITLE
Rule blocked: CG0014

### ISSUE THREAD

**Issue by gerrycampion** [cdisc-org/cdisc-rules-engine#484]
**Links to related JIRA Tickets**
- https://jira.cdisc.org/projects/CORERULES/issues/CORERULES-
- https://jira.cdisc.org/projects/CORERULES/issues/CORERULES-

**Rule Information**
- **Standard**: SDTMIG
- **Rule ID**: CG0014 (A)
- **Rule Description**:
Variable: GEN
Condition: Variable Core Status = Required
Rule: Variable present in dataset _( and ^= null ) <-- Not included in Part A_

**Describe the bug**
`not_contains_all` operator not working correctly with `get_column_order_from_dataset` operation.
Refer to CG0014 (A) in the rule editor and corresponding negative test data.

```yaml
Rule Type: Dataset Metadata Check
Operations:
  - id: $required_variables
    operator: required_variables
  - id: $dataset_variables
    operator: get_column_order_from_dataset
Check:
  all:
    - name: $dataset_variables
      operator: not_contains_all
      value:
        - $required_variables
```

`get_column_order_from_dataset` appears to return a list object, but `not_contains_all` expects `self.value[target]` to be a series object in order to apply `unique()`.

**Error returned from Rule Engine**
```json
{
  "error": "An unknown exception has occurred",
  "message": "unhashable type: 'list'"
}
```

**Expected behavior**
No errors. The rule should return no issues when all required variables exist or return an issue for missing required variables.

**Comment by nhaydel** [comment]
@gerrycampion why not just use variable_name as the target instead with a rule type of Variable Metadata Check:

i.e:

```
Check:
  all:
    - name: variable_name
      operator: not_contains_all
      value:
        - $required_variables
```

I get that we may want to fix this, but is the rule actually blocked by this bug?

**Comment by gerrycampion** [comment]
No, that should work, thanks. I updated the issue title

**Comment by ab3263266** [comment]
Can I ask variable_name gives the list of all the variables in the dataset? If yes it is very unclear

**Comment by gerrycampion** [comment]
> Can I ask variable_name gives the list of all the variables in the dataset? If yes it is very unclear

[variable_name](https://cdisc-org.github.io/conformance-rules-editor/#/metadata_variables?id=variable_name) is one of the metadata variables available when using the [Variable Metadata Check](https://cdisc-org.github.io/conformance-rules-editor/#/rule_type?id=variable-metadata-check) rule type.

The `Variable Metadata Check` rule type produces a data frame containing one record for each variable in the current dataset. At the dataset level, you are correct that `variable_name` gives the list of all variables in the dataset. This means that at the record level, `variable_name` gives the name of the variable of the current record.

@ab3263266

**Comment by SFJohnson24** [comment]
this is resolved and the rule is published
